### PR TITLE
feat(lib) Add deepOmit utility function

### PR DESCRIPTION
**Pull request by TheSonOfThomp**
Introduce a new utility function `deepOmit` that allows for deeply omitting properties from objects based on specified paths. This addition includes comprehensive tests to ensure functionality and edge case handling.

### CODE REVIEW

File: packages/lib/src/helpers/deepOmit/deepOmit.spec.ts [hunk 1]
```diff
@@ -0,0 +1,314 @@
+import { deepOmit } from './deepOmit';
+
+describe('deepOmit', () => {
```
Comment (stephl3) [inline packages/lib/src/helpers/deepOmit/deepOmit.spec.ts:3]: have you considered adding a test case for duplicate keys that are adjacent or nested?

Comment (TheSonOfThomp) [reply packages/lib/src/helpers/deepOmit/deepOmit.spec.ts:3]: No, since that wouldn't be allowed as an input object in JS 
```js
const x = { a: 1, a: 2 } // { a: 2 }
```

Comment (stephl3) [reply packages/lib/src/helpers/deepOmit/deepOmit.spec.ts:3]: Ahh misspoke, I meant like this case:
```
const x = { a: { b: 1 }, b: 2 };
```
